### ISSUE TITLE
Angular with Polymer Containers

### ISSUE THREAD

**Issue by JesterXL** [webcomponents/webcomponentsjs#396]
If you use Angular on top of Polymer containers, it only seems to work in Chrome when you switch routes HTML5 mode disabled using ui-router. Deep linking there works, but navigating in browser, the DOM method 'after' looks for a `parentNode` that isn't there. In Chrome, I guess either the polyfill isn't needed, or in Firefox/Safari, it fails to see the parentNode, and instead sets this `_lightParent` property.

If I hack Angular's after method from:

``` javascript
after: function(element, newElement) {
    var index = element, parent = element.parentNode;
    newElement = new JQLite(newElement);

    for (var i = 0, ii = newElement.length; i < ii; i++) {
      var node = newElement[i];
      parent.insertBefore(node, index.nextSibling);
      index = node;
    }
  }
```

To this:

``` javascript
after: function(element, newElement) {
    var index = element;
    var parent = element.parentNode || element._lightParent;
    newElement = new JQLite(newElement);

    for (var i = 0, ii = newElement.length; i < ii; i++) {
      var node = newElement[i];
      parent.insertBefore(node, index.nextSibling);
      index = node;
    }
  }
```

That seems to fix it in Firefox and Safari. Maybe this is fixed in Angular 2 with their knowledge of ShadowDOM, but I need something today, and Angular is too hard to monkey-patch.

If you could explain what the problem is more clearly, I could help do a PR.

**Comment by addyosmani** [comment]
Reviewing the samples supplied (thanks!), it's unclear whether this is an issue with Angular 1 core (learning in that direction) or the polyfills. Some follow-up questions:

* Are you using webcomponents.js or webcomponents-lite.js? (Polymer 1.0 generally is typically used with the latter, unless you're intentionally shimming ShadowDOM)
* Does this work with the polyfills disabled in Chrome? (confirms it likely isn't a polyfill issue)

Thanks!

**Comment by JesterXL** [comment]
1. webcomponents.js
2. No, I only know how to turn those off in Dart, not JavaScript, heh!

If I put the main Polymer container BACK into a dom-module, and expose both content and ng-transclude and ui-view... everyone seems to be happy.
https://gist.github.com/JesterXL/bf8156a61df7704c4e4a

**Comment by JesterXL** [comment]
Just tried with webomponents-lite.js, same thing. Firefox and Safari are fine if I deep link, but as soon as I change from server.com to server.com/#/login, it throws that exception.

**Comment by octavioamu** [comment]
having this issue too

**Comment by es6Test** [comment]
What does polymer add on top of this library?

**Comment by kasperstorgaard** [comment]
same issue here

**Comment by nathantsoi** [comment]
what exception does it throw?

i had a possibly related situation where polymer  + angular would work fine until it got to polymer content tag.

turns out, at least when manually bootstrapping, i needed to specify the main document, wrapped, otherwise it would throw a bunch of errors as it compiled the dom (in coffeescript):

```coffee
 angular.element(document).ready ->
-  angular.bootstrap(document, ['myawesomeapp'])
+  doc =
+    if window.ShadowDOMPolyfill
+      window.ShadowDOMPolyfill.wrapIfNeeded(document)
+    else
+      document
+  angular.bootstrap(doc, ['myawesomeapp']
```

**Comment by TimvdLippe** [comment]
Thanks for the issue. It has been a great while since this issue was opened. Sadly, this issue misses a live reproduction with the most recent version of the webcomponents polyfill for us to be able to debug and potentially fix. Versions 1 of the polyfills fix a large amount of issues, probably including this one as well.

Please see [the CONTRIBUTING.md](https://github.com/webcomponents/webcomponentsjs/blob/master/CONTRIBUTING.md) for guidelines on filing issues.  You can start from [this jsBin template](http://jsbin.com/birajez/edit?html,console,output) as a starting point and link it in the issue template under "Live Demo".

Feel free to file a new issue following these instructions if you can still reproduce this issue with versions 1 of the polyfills.  Thank you!
